src/Algebra/Graph/AdjacencyIntMap/Algorithm.hs

```diff
@@ -213,32 +243,86 @@ dfs vs = concatMap flatten . dfsForestFrom vs
 -- 'isSubgraphOf' ('vertices' $ reachable x y) y == True
 -- @
 reachable :: Int -> AdjacencyIntMap -> [Int]
-reachable x = concat . bfs [x]
+reachable x = dfs [x]
+
+type Cycle = NonEmpty
+data NodeState = Entered | Exited
+data S = S { table :: IntMap.IntMap Int
```

Comment:
Could we give a more meaningful name to `table`? What do the entries of the table correspond to?

Reply:
The entries correspond to edges where `table ! u = v` means that `v` is `u`'s parent in the search tree. I changed the name to `parent` if that works?

Reply:
Yes, `parent` is great!